File: public-api/src/main/java/com/shopleech/publicapi/dal/repository/ShopRepository.java
package com.shopleech.publicapi.dal.repository;

import com.shopleech.publicapi.domain.Shop;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.stereotype.Repository;

import java.util.List;

/**
 * @author Ahto Jalak
 * @since 30.03.2023
 */
@Repository
public interface ShopRepository extends JpaRepository<Shop, Integer>, ShopRepositoryCustom {
    @Query(value = "SELECT COUNT(*) FROM shop", nativeQuery = true)
    int countAll();

    @Query(value = "SELECT * FROM shop where name ilike '%'||:name||'%'", nativeQuery = true)
    List<Shop> findByName(String name);
}

File: price-comparison-client1/src/dal/repository/ShopRepository.ts
import { BaseRepository } from '@/dal/repository/BaseRepository'
import type { IServiceResult } from '@/bll/service/model/IServiceResult'
import type { IShopRepository } from '@/dal/repository/model/IShopRepository'
import httpCLient from '@/util/http-client'
import { AxiosError } from 'axios'
import type { IShop } from '@/dal/domain/IShop'

/**
 * @author Ahto Jalak
 * @since 12.04.2023
 */
export class ShopRepository extends BaseRepository<IShop> implements IShopRepository {
    constructor () {
        super('v1/shop')
    }

    async findByName (entity: IShop): Promise<IServiceResult<IShop[]>> {
        this.logger.info('findByName')

        let response
        try {
            response = await httpCLient.post(`/${this._path}/search`, entity, {
                headers: {
                    Authorization: 'Bearer ' + this.identityStore.$state.jwt?.token,
                }
            })
        } catch (e:any) {
            return {
                status: (e as AxiosError).response?.status,
                errorMsg: e.response?.data.message,
            }
        }

        return {
            status: response.status,
            data: response.data.details,
        }
    }
}
